lua/ttt2/libraries/migrations.lua

```diff
@@ -0,0 +1,184 @@
+---
+-- A TTT2 version migrations library
+-- @author ZenBre4ker
+-- @module migrations
+
+if SERVER then
+	AddCSLuaFile()
+end
+
+migrations = {}
+migrations.orm = nil
+migrations.databaseName = "ttt2_migrations"
+migrations.savingKeys = {
+	date = {
+		typ = "string",
+		default = ""
+	},
+	wasSuccess = {
+		typ = "bool",
+		default = false
+	},
+	wasUpgrade = {
+		typ = "bool",
+		default = true
+	},
+}
+migrations.folderPath = "terrortown/migrations/"
+migrations.commands = {}
+migrations.cachedCommands = {}
+migrations.isLoaded = false
+
+---
+-- Get ORM table and create it if not available yet
+-- @return @{ORMOBJECT} The orm-table to address the database
+-- @realm shared
+function migrations.GetORM()
+	if istable(migrations.orm) then
+		return migrations.orm
+	end
+
+	-- Create Sql and orm table if not already done or savingKeys were changed
+	sql.CreateSqlTable(migrations.databaseName, migrations.savingKeys)
+	migrations.orm = orm.Make(migrations.databaseName)
+
+	return migrations.orm
+end
+
+---
+-- Creates a command with state information, an up- and downgrade function
+-- @param table states The states that are needed to fulfill the upgrade
+-- @param function upgrade The function that is used for forward operations
+-- @param function downgrade The function that is used for reverting operations of this command
+-- @return @{command} A command containing necessary functionality for forward and backward operations
+-- @realm shared
+function migrations.CreateCommand(states, upgrade, downgrade)
+	if not istable(states) or not isfunction(upgrade) or not isfunction(downgrade) then
+		ErrorNoHalt("[TTT2] Couldn't create migrations command. Missing states-table, upgrade- or downgrade-function.\n")
+		states = {}
+		upgrade = function(cmd) return end
+		downgrade = function(cmd) return end
+	end
```

Comment:
should we really create an empty migration here? This should only ever happen in dev imo

Reply:
Maybe a command triggering an error inside the migration is better?

We somehow need to help us developing migrations and make sure they are complete.